integrations/extensions/starter-kits/watson-discovery/README.md

```diff
@@ -102,9 +103,13 @@ You can also go to the "Preview" tab of Watson Assistant tooling and see what th
+To explore the actions, go to the Actions page in Watson Assistant.  You may want to start with the "Set by assistant" list of actions and click on "No action matches".  The first step of this action calls out to the "Search" action (which is in the "Created by you" list of actions).  If you have an assistant for which you expect search to be common and important, you may want to end your "No action matches" action there, so that when no action matches, you will _only_ get search results.  However, in our starter kit, we've included some extra logic that might be useful for assistants where search results are expected to be infrequent and less relevant.  Specifically, we check to see if "No action matches" has triggered 3 or more times.  If not, we end the action, but if so, we also trigger an "escalate to agent" event, which gives the user an option to connect directly to your service desk (if you have configured one).  The motivation here is that if a user has issued that many queries that did not match any of your actions and only produced search results, they may not be getting the information they need and may need more help.  Of course, if you do not have a service desk configured, then you should get rid of this logic and terminate after the first step.
+
+As noted above, "No action matches" calls out to "Search" in "Created by you".  The first step of "Search" sets the `query_text` to `input.original_text`.  As noted in the [documentation for spell checking](https://cloud.ibm.com/docs/assistant?topic=assistant-dialog-runtime-spell-check#dialog-runtime-spell-check-how-it-works), `input.original_text` is set _only_ if the utterance from the user was altered due to spell correction and then it records the original request from the user and not the spell corrected text.  Spell correction can be very counter productive for searching because it can take specialized domain vocabulary and "correct" those terms to generic words in the language, so it is often better to apply the search on the original text.  The next step in Search then checks to see if the `query_text` is set (which it will be only if spell correction activated) and if not, it falls back to `input.text`.  At this point, `query_text` is guaranteed to be the exact original query issued by the user.  The next calls the Watson Discovery search extension.  The next step provides an error message if the search failed.  The following step then calls out to the "Show search results" action.  The final step then empties out both the result list in the extension response and the search results variable.  This is done to avoid keeping too much data in memory; see the "Limit on Size of Search Results" section later in this document for more details on why this is important.
```

Comment:
```suggestion
As noted above, "No action matches" calls out to "Search" in "Created by you".  The first step of "Search" sets the `query_text` to `input.original_text`.  As noted in the [documentation for spell checking](https://cloud.ibm.com/docs/assistant?topic=assistant-dialog-runtime-spell-check#dialog-runtime-spell-check-how-it-works), `input.original_text` is set _only_ if the utterance from the user was altered due to spell correction and then it records the original request from the user and not the spell corrected text.  Spell correction can be very counter productive for searching because it can take specialized domain vocabulary and "correct" those terms to generic words in the language, so it is often better to apply the search on the original text.  The next step in Search then checks to see if the `query_text` is set (which it will be only if spell correction activated) and if not, it falls back to `input.text`.  At this point, `query_text` is guaranteed to be the exact original query issued by the user.  The next step calls the Watson Discovery search extension.  The next step provides an error message if the search failed.  The following step then calls out to the "Show search results" action.  The final step then empties out both the result list in the extension response and the search results variable.  This is done to avoid keeping too much data in memory; see the "Limit on Size of Search Results" section later in this document for more details on why this is important.
```